service/manager.go

```diff
@@ -331,26 +331,28 @@ func (m *manager) start() Control {
 		m.registerSignalHandlers()
 		if len(errs) > 0 {
 			var serviceErr error
-			errChan := make(chan Exit, 1)
-			// grab the first error, shut down the service and return the error
+
+			// emit all errors over the errChan channel
+			errChan := make(chan Exit, len(errs))
 			for _, e := range errs {
 				errChan <- Exit{
 					Error:    e,
 					Reason:   "Module start failed",
 					ExitCode: 4,
 				}
-
-				m.shutdownMu.Unlock()
-				if _, err := m.shutdown(e, "", nil); err != nil {
-					zap.L().Error("Unable to shut down modules",
-						zap.NamedError("initialError", e),
-						zap.NamedError("shutdownError", err),
-					)
-				}
 				zap.L().Error("Error starting the module", zap.Error(e))
-				// return first service error
+
+				// if it is the first error, shut down the service
 				if serviceErr == nil {
```

Comment:
One other thing might be more clear is to take the whole thing out of `for` loop and right after `if len(errs) > 0`. Just assign the first `err` to `serviceErr`

Doing assignment and checkings in a look is kinda mind twisting

Reply:
Good point, cleaned up